packages/shared/client/ReplayClient.ts

```diff
@@ -981,12 +981,10 @@ export class ReplayClient implements ReplayClientInterface {
         const loadedRegions = this.loadedRegions;
         let isLoaded = false;
         if (loadedRegions !== null) {
-          if (focusRange !== null) {
-            isLoaded = isRangeInRegions(focusRange, loadedRegions.loaded);
+          if (focusRange === null) {
```

Comment:
I switched the order of these branches, it felt like if we are going to have an `else` clause anyways, then removing negatives slightly reduces cognitive load while reading, but I don't feel strongly about it.

Reply:
I don't really feel strongly about one vs another– but this feels pretty subjective to me. I'd prefer we don't make changes like this often, because it seems like it would be a lot of churn for a subjective prefernce.

Reply:
I agree about the diff noise created by things that are just based on preference. I really do think that the `!=` operator causes more cognitive load than the `==` operator, but I don't have proof that this is the case.

FWIW, I can find some people on the internet who agree:

- https://csharpdeveloper.wordpress.com/2016/01/20/roberts-rules-of-coders-8-avoid-negative-conditionals-and-method-names/
- https://schneide.blog/2014/08/03/dont-ever-not-avoid-negative-logic/

But if that were a convincing argument we would have larger problems so...

Reply:
> FWIW, I can find some people on the internet who agree

lol 😆

Reply:
Oh, funny, I 100% agree when the truthiness check is used, and I disagree when the explicit `!== null` or `!= null` is used. For me it's two things that make it difficult: `undefined !== null` evaluates to `true`, so even though TS is helping, I still hold that possibility in my head just in case we bungled up our types somewhere, whereas `=== null` or `== null` alleviates that (but so does `if (thing)`. The rubyist in me really wants `if (thing)`, but I know that's not anyone's cup of tea. Anyways, I'll happily cede to you hear, thanks for explaining your thought process!

Reply:
Unfortunately `!= null` and `== null` are generally better options, because `if (thing)` is often _not going to give the result you want_ because of "falsey" values like 0 and `""` (empty string)

Reply:
Heh. And on my side, I've internalized "never use the loose single-`=` comparison operators, ever" :)  so seeing Brian frequently use `!=` has been kinda jarring.

Reply:
Curious why you've internalized that?

My reasoning for it is mostly just that... it's easy for `undefined` to slip into the chain unintentionally (e.g. `map.get(key)`) and _in most cases_ `null` and `undefined` have similar semantic meaning (unlike other "falsey" values).